mopidy_party/__init__.py

```diff
@@ -24,22 +27,64 @@ def get(self):
             self.data["track"] = currentTrackURI
             self.data["votes"] = []
 
-        if (self.request.remote_ip in self.data["votes"]): # User has already voted
+        if (self._getip() in self.data["votes"]): # User has already voted
             self.write("You have already voted to skip this song =)")
         else: # Valid vote
-            self.data["votes"].append(self.request.remote_ip)
+            self.data["votes"].append(self._getip())
             if (len(self.data["votes"]) == self.requiredVotes):
                 self.core.playback.next()
                 self.write("Skipping...")
             else:
                 self.write("You have voted to skip this song. ("+str(self.requiredVotes-len(self.data["votes"]))+" more votes needed)")
 
 
+class AddRequestHandler(tornado.web.RequestHandler):
+
+    def initialize(self, core, data, config):
+        self.core = core
+        self.data = data
+
+    def _getip(self):
+        return self.request.headers.get("X-Forwarded-For", self.request.remote_ip)
+
+    def post(self):
+        # when the last n tracks were added by the same user, abort.
+        if self.data["queue"] and all([e == self._getip() for e in self.data["queue"]]):
```

Comment:
Maybe it's on purpose but just in case: if there is no song in the queue, and one user has already added max_songs, they cannot add any music anymore.